src/EventListener/DoctrineListener.php

```diff
@@ -82,12 +84,52 @@ public function postConnect(ConnectionEventArgs $args)
             // the outcome of a GROUP_CONCAT() query will be more than 1024 bytes.
             // See also: http://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_group_concat_max_len
             $db->executeQuery('SET SESSION group_concat_max_len = 100000');
-        } elseif ($platform === 'postgresql') {
-            /**
-             * @see https://github.com/doctrine/dbal/pull/828
-             */
+        } elseif ($platformName === 'postgresql') {
+            /** @see https://github.com/doctrine/dbal/pull/828 */
             $db->executeQuery("SET NAMES 'utf8'");
         }
+
+        if (DBAL\Version::compare('2.6.0' >= 0)) {
+            $this->compatDbalPre26($platform, $platformName);
+        }
+    }
+
+    /**
+     * Cross-platform backwards compatibility for JSON field types.
+     *
+     * DBAL 2.6 deprecated JsonArrayType in favour of JsonType, and bumped the
+     * minimum PHP version to 7.1. As we need to maintain support for PHP 5.5+
+     * this has caused problems with schema change detection.
+     *
+     * @see https://github.com/bolt/bolt/issues/6856
+     *
+     * @deprecated Remove when minimum PHP version is 7.1 or later.
+     *
+     * @param Platforms\AbstractPlatform $platform
+     * @param string                     $platformName
+     */
+    private function compatDbalPre26(Platforms\AbstractPlatform $platform, $platformName)
+    {
+        if ($platformName === 'sqlite') {
+            /** @see \Doctrine\DBAL\Platforms\SqlitePlatform::getClobTypeDeclarationSQL */
+            $platform->registerDoctrineTypeMapping('CLOB', Schema\Types\JsonType::JSON);
+        } elseif ($platformName === 'mysql') {
+            if ($platform instanceof Platforms\MySQL57Platform) {
+                /** @see \Doctrine\DBAL\Platforms\MySQL57Platform::getJsonTypeDeclarationSQL */
+                $platform->registerDoctrineTypeMapping('JSON', Schema\Types\JsonType::JSON);
+            } else {
+                /** @see \Doctrine\DBAL\Platforms\MySqlPlatform::getClobTypeDeclarationSQL */
+                $platform->registerDoctrineTypeMapping('LONGTEXT', Schema\Types\JsonType::JSON);
+            }
+        } elseif ($platformName === 'postgresql') {
+            if ($platform instanceof Platforms\PostgreSQL92Platform) {
+                /** @see \Doctrine\DBAL\Platforms\PostgreSQL92Platform::getJsonTypeDeclarationSQL */
+                $platform->registerDoctrineTypeMapping('JSON', Schema\Types\JsonType::JSON);
+            } else {
+                /** @see \Doctrine\DBAL\Platforms\PostgreSqlPlatform::getClobTypeDeclarationSQL */
+                $platform->registerDoctrineTypeMapping('TEXT', Schema\Types\JsonType::JSON);
+            }
+        }
```

Comment:
This should be moved to `JsonType::getMappedDatabaseTypes`.

Reply:
Moved